src/Kestrel.Core/Internal/ConnectionHandler.cs

```diff
@@ -34,89 +28,45 @@ public void OnConnection(IFeatureCollection features)
 
             var transportFeature = connectionContext.Features.Get<IConnectionTransportFeature>();
 
-            var inputPipe = transportFeature.PipeFactory.Create(GetInputPipeOptions(transportFeature.InputWriterScheduler));
-            var outputPipe = transportFeature.PipeFactory.Create(GetOutputPipeOptions(transportFeature.OutputReaderScheduler));
+            // REVIEW: Unfortunately, we still need to use the service context to create the pipes since the settings
+            // for the scheduler and limits are specified here
```

Comment:
Could you just pass in the KestrelServerOptions? Do you need to pass in a scheduler and the "ThreadPool"?